Google.PowerShell/Container/GkeClusterCmdlets.cs

```diff
@@ -1136,6 +1011,560 @@ private void PopulateClusterNetwork(Cluster cluster)
+        /// per cmdlet invocation.
+        /// </summary>
+        private void UpdateNodePool(ClusterUpdate clusterUpdate)
+        {
+            string imageType = null;
+            string nodeVersion = null;
+
+            if (_dynamicParameters.ContainsKey("ImageType"))
+            {
+                imageType = _dynamicParameters["ImageType"].Value?.ToString().ToLower();
+                if (!string.IsNullOrWhiteSpace(imageType))
+                {
+                    // Have to check that -NodeVersion and -Min/MaxNodesToScaleTo are not set
+                    // because the cmdlet only does one update.
+                    if (!string.IsNullOrWhiteSpace(nodeVersion) || MaximumNodesToScaleTo != null
+                        || MininumNodesToScaleTo != null)
+                    {
+                        throw new PSArgumentException("When -ImageType is used, -NodeVersion" +
+                            " and -Max/MinNodesToScaleTo cannot be used.");
+                    }
+                    clusterUpdate.DesiredImageType = imageType;
+                    return;
+                }
+            }
+
+            if (_dynamicParameters.ContainsKey("NodeVersion"))
+            {
+                nodeVersion = _dynamicParameters["NodeVersion"].Value?.ToString().ToLower();
+                if (!string.IsNullOrWhiteSpace(nodeVersion))
+                {
+                    // If we reach here, then it is for sure that -ImageType is either null or
+                    // empty so we don't have to check it.
+                    if (MaximumNodesToScaleTo != null || MininumNodesToScaleTo != null)
+                    {
+                        throw new PSArgumentException("When -NodeVersion is used, -ImageType" +
+                            " and -Max/MinNodesToScaleTo cannot be used.");
+                    }
+
+                    UpdateNodeVersion(clusterUpdate, nodeVersion);
+                    return;
+                }
+            }
+
+            if (MaximumNodesToScaleTo == null)
+            {
+                throw new PSArgumentException("Either -ImageType, -NodeVersion" +
+                    " or -Max/MinNodesToScaleTo should be used when -NodePoolName is used.");
+            }
+
+            UpdateAutoScaling(clusterUpdate);
+        }
+
+        /// <summary>
+        /// Helper function to set the DesiredNodeVersion in clusterUpdate
+        /// based on the string nodeVersion. This function also performs check
+        /// to make sure that the node version is less than the master version.
+        /// </summary>
+        private void UpdateNodeVersion(ClusterUpdate clusterUpdate, string nodeVersion)
+        {
+            // We have to make sure that the node version we are updating to is less than
+            // that of the master.
+            Version resolvedNodeVersion = null;
+            if (string.Compare(nodeVersion, "latest", true) != 0)
```

Comment:
I guess `"latest".Equals(nodeVersion, StringComparison.OrdinalIgnoreCase)` is too long?